src-self-hosted/main.zig

```diff
@@ -687,19 +687,61 @@ const usage_fmt =
     \\Options:
     \\   --help                 Print this help and exit
     \\   --color [auto|off|on]  Enable or disable colored error messages
-    \\
+    \\   -r                     Recursive (searches within directories)
     \\
 ;
 
 const args_fmt_spec = []Flag{
     Flag.Bool("--help"),
+    Flag.Bool("-r"),
     Flag.Option("--color", []const []const u8{
         "auto",
         "off",
         "on",
     }),
 };
 
+// Returns false if any errors occur
+fn fmtHandleFile(allocator: *Allocator, file_path: []const u8, color: errmsg.Color) !bool {
+    var file = try os.File.openRead(allocator, file_path);
+    defer file.close();
+
+    return blk: {
+        const source_code = io.readFileAlloc(allocator, file_path) catch |err| {
+            try stderr.print("unable to open '{}': {}\n", file_path, err);
+            break :blk false;
```

Comment:
You can just use a straight return for these. No need to pass return via the `:blk`.